source/simulator/core.cc

```diff
@@ -2012,6 +2015,10 @@ namespace aspect
           last_checkpoint_time = std::time(nullptr);
 
         if (should_terminate)
+          pcout << "   Total wallclock time elapsed including restarts:" << round(walltime.wall_time()+walltime_from_resume) << "s" << std::endl;
+          walltime_from_resume += walltime.wall_time();
+          walltime.reset();
+          // serialize walltime_from_resume     
```

Comment:
I moved these 4 lines from checkpoint_restart as we don't need to print or restart the timer every checkpoint, only on exiting. But since moving it here when I test it locally a test model only runs for the first time step and says it's completed when it should run for many years..

Reply:
The if statement body only covers the first statement (up to the first semicolon). This is clearly wrong.
If you indent the code using make indent, you would have noticed your mistake.

Reply:
To explain: right now the "break" is outside the if, so you will always terminate.

Reply:
Ok thanks. I now realise this is not the place for the print statement and I've moved it to follow the instances of "computing_timer.print_summary ();". However, with a current test the deal table tells me the total wall clock time is 436s and the first print statement we implement here says only 65s so the wallclock.restart is in the wrong place?